### PR TITLE
Amazon-AWS-EC2-Snapshots3

### PR DESCRIPTION
**Pull request by mihaitusa**
added sl's && test sl's

### CODE REVIEW

**Comment by mihaitusa** [comment]
We will merge it after release to MC on: CloudSlang/score-actions#133

**Comment by mihaitusa** [comment]
@Bonczidai Please review it!

**Comment by mihaitusa** [comment]
@Bonczidai Please review it!

File: content/io/cloudslang/cloud/amazon_aws/snapshots/create_snapshot_in_region.sl [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    - proxy_port:
+        required: false
+    - proxyPort:
+        default: ${get("proxy_port", "8080")}
+        private: true
+    - region:
+        default: 'us-east-1'
+        required: false
+    - volume_id
+    - volumeId: ${volume_id}
```
Comment (Bonczidai) [inline content/io/cloudslang/cloud/amazon_aws/snapshots/create_snapshot_in_region.sl]: does this need to be private?

File: content/io/cloudslang/cloud/amazon_aws/snapshots/create_snapshot_in_region.sl [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+#! @input provider: Cloud provider on which the instance is - Default: 'amazon'
+#! @input endpoint: Endpoint to which first request will be sent - Default: 'https://ec2.amazonaws.com'
+#! @input identity: optional - Amazon Access Key ID
+#! @input credential: optional - Amazon Secret Access Key that corresponds to the Amazon Access Key ID
+#! @input proxy_host: optional - Proxy server used to access the provider services
+#! @input proxy_port: optional - Proxy server port used to access the provider services - Default: '8080'
+#! @input region: optional - Region where volume to make a snapshot for it reside. ListRegionAction can be used in order
+#!                           to get all regions. For further details check: http://docs.aws.amazon.com/general/latest/gr/rande.html#s3_region
+#!                         - Default: 'us-east-1'
+#! @input volumeId: ID of the EBS volume
```
Comment (Bonczidai) [inline content/io/cloudslang/cloud/amazon_aws/snapshots/create_snapshot_in_region.sl]: snake case?

File: content/io/cloudslang/cloud/amazon_aws/snapshots/create_snapshot_in_region.sl [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+#!               that are created from encrypted snapshots are also automatically encrypted. Your encrypted volumes and
+#!               any associated snapshots always remain protected. For more information, see Amazon Elastic Block Store
+#!               and Amazon EBS Encryption in the Amazon Elastic Compute Cloud User Guide.
+#! @input provider: Cloud provider on which the instance is - Default: 'amazon'
+#! @input endpoint: Endpoint to which first request will be sent - Default: 'https://ec2.amazonaws.com'
+#! @input identity: optional - Amazon Access Key ID
+#! @input credential: optional - Amazon Secret Access Key that corresponds to the Amazon Access Key ID
+#! @input proxy_host: optional - Proxy server used to access the provider services
+#! @input proxy_port: optional - Proxy server port used to access the provider services - Default: '8080'
+#! @input region: optional - Region where volume to make a snapshot for it reside. ListRegionAction can be used in order
```
Comment (Bonczidai) [inline content/io/cloudslang/cloud/amazon_aws/snapshots/create_snapshot_in_region.sl]: maybe refactor this sentence ~ Region where the volume, we want to create a snapshot for, resides

File: content/io/cloudslang/cloud/amazon_aws/snapshots/delete_snapshot_in_region.sl [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+  inputs:
+    - provider: 'amazon'
+    - endpoint: 'https://ec2.amazonaws.com'
+    - identity:
+        default: ''
+        required: false
+        sensitive: true
+    - credential:
+        default: ''
+        required: false
```
Comment (Bonczidai) [inline content/io/cloudslang/cloud/amazon_aws/snapshots/delete_snapshot_in_region.sl]: if it's required false, you don't need to specify an empty default - IMHO it's the inner logic's responsability to treat this case

**Comment by Bonczidai** [comment]
maybe add `sensitive` modifier in flow description

:+1: looks good, please handle comments

File: content/io/cloudslang/cloud/amazon_aws/snapshots/create_snapshot_in_region.sl (hunk 1 above)
Comment (mihaitusa) [inline content/io/cloudslang/cloud/amazon_aws/snapshots/create_snapshot_in_region.sl]: yes... the user should not have access to it ... only through proxy_port input

File: content/io/cloudslang/cloud/amazon_aws/snapshots/delete_snapshot_in_region.sl (hunk 4 above)
Comment (mihaitusa) [inline content/io/cloudslang/cloud/amazon_aws/snapshots/delete_snapshot_in_region.sl]: almost 99,99 % of the Amazon users have credentials... so, I think we can let it as it is... WDYT?

File: content/io/cloudslang/cloud/amazon_aws/snapshots/create_snapshot_in_region.sl (hunk 2 above)
Comment (mihaitusa) [inline content/io/cloudslang/cloud/amazon_aws/snapshots/create_snapshot_in_region.sl]: @Bonczidai  yes... you are right!

File: content/io/cloudslang/cloud/amazon_aws/snapshots/create_snapshot_in_region.sl (hunk 3 above)
Comment (mihaitusa) [inline content/io/cloudslang/cloud/amazon_aws/snapshots/create_snapshot_in_region.sl]: thanks @SamMarkowitz !

**Pull request merged**